project/src/main/kotlin/org/ionproject/core/programme/ProgrammeServices.kt

```diff
@@ -5,7 +5,7 @@ import org.ionproject.core.common.model.ProgrammeOffer
 import org.springframework.stereotype.Component
 
 @Component
-class ProgrammeServices(private val repo : ProgrammeRepo) {
+class ProgrammeServices(private val repo : ProgrammeRepoImpl) {
```

Comment:
The previous version was correct. The spring runtime will create an instance of a class which will implement the interface and inject it here. **ANY** instance can be injected here, as long as it implements the interface. Spring will look for the highest priority Component, and use it.

During testing, we may want to create a MockProgrammeRepo which will implement the interface and which will be configured as the higher priority Component (overriding the ProgrammeRepoImpl component) at testing time.

_**TLDR: by placing ProgrammeRepoImpl here (a non-open class bear in mind), a MockProgrammeRepo cannot be used instead for testing.**_

Reply:
The tests has mentioned by the project coordinator will be in a different database with mock data, isn't it irrelevant to have 2 different repositories for that?